docs/ml_system_design_doc.md

```diff
@@ -0,0 +1,201 @@
+- Для бейзлайна: Генерация персонажей по фиксированным шаблонам без адаптивного поведения.
+- Метрики качества: Уникальность персонажей (не менее 80% различий в ключевых характеристиках), согласованность поведения персонажа с его характеристиками (оценка экспертами, не менее 7/10).
+- Риски: Шаблонность персонажей, несогласованность поведения. Планируем использовать расширенные наборы характеристик и проверку внутренней согласованности.
+
+*Этап 5 - Разработка агентов для управления сюжетом*
+
+- Для MVP: Исследование и создание набора агентов для генерации сюжетных линий с учетом мира и персонажей. Реализация механизмов для адаптации сюжета к действиям пользователя.
+- Для бейзлайна: Генерация линейного сюжета без возможности ветвления.
+- Метрики качества: Адаптивность сюжета (количество значимых изменений в ответ на действия пользователя, не менее 5 на сессию), увлекательность (оценка пользователями, не менее 7/10).
+- Риски: Линейность сюжета, предсказуемость. Планируем использовать динамические сюжетные шаблоны и механизмы неожиданных поворотов.
+
+*Этап 6 - Разработка агентов для взаимодействия с пользователем*
+
+- Для MVP: Исследование и создание набора агентов для обработки ввода пользователя и интеграции его действий в мир и сюжет. Реализация базового набора команд и действий.
+- Для бейзлайна: Обработка ограниченного набора предопределенных команд.
+- Метрики качества: Процент успешно распознанных команд пользователя (не менее 85%), естественность ответов системы (оценка пользователями, не менее 7/10).
+- Риски: Ограниченное понимание естественного языка, шаблонные ответы. Планируем расширять набор понимаемых команд на основе тестирования.
+
+*Этап 7 - Интеграция и тестирование*
+
+- Для MVP и бейзлайна: Интеграция всех компонентов, проведение внутреннего тестирования, устранение ошибок и несогласованностей.
+- Метрики качества: Стабильность системы (не более 1 сбоя на 100 взаимодействий), общая оценка пользовательского опыта (не менее 7/10).
+- Риски: Интеграционные проблемы, несогласованность между агентами. Планируем поэтапное тестирование и итеративное улучшение.
+
+### 3. Подготовка пилота  
+  
+#### 3.1. Способ оценки пилота  
+
+- Тестовой группе из 20-30 пользователей будет предоставлен доступ к системе на 2 недели. Пользователи будут создавать персонажей и проходить сгенерированные приключения. После каждой сессии будут собираться количественные метрики (время сессии, количество взаимодействий) и качественная обратная связь через опросники. Также будут проведены интервью с частью пользователей для получения более детальной обратной связи.
+
+#### 3.2. Что считаем успешным пилотом  
+
+- Количественные метрики:
+  - Не менее 70% пользователей проводят в системе более 30 минут за сессию
```

Comment:
Is it a representative metric? If your system generates a great plot from the first attempt, and very fast, then you likely will not achieve this metric.